s3/s3.go

```diff
@@ -1 +1,100 @@
 package s3
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/credentials"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/s3"
+	"github.com/opensaucerer/bifrost/shared/errors"
+	"github.com/opensaucerer/bifrost/shared/types"
+)
+
+func (s SimpleStorageService) UploadFile(path, filename string) (types.UploadedFile, error) {
+	var ctx context.Context
+	var cancel context.CancelFunc
+	ctx = context.Background()
+	if s.DefaultTimeout > 0 {
+		ctx, cancel = context.WithTimeout(ctx, time.Duration(s.DefaultTimeout)*time.Second)
+		defer cancel()
+	}
+	// verify that file exists
+	if _, err := os.Stat(path); os.IsNotExist(err) {
+		return types.UploadedFile{}, &errors.BifrostError{
+			Err:       fmt.Errorf("file does not exist: %s", path),
+			ErrorCode: errors.ErrBadRequest,
+		}
+	}
+	// Create an AWS session
+	sess, err := session.NewSession(&aws.Config{
+		Region:      aws.String(s.Region),
+		Credentials: credentials.NewStaticCredentials(s.AccessKey, s.SecretKey, "")},
+	)
+	if err != nil {
+		return types.UploadedFile{}, &errors.BifrostError{
+			Err:       fmt.Errorf("error creating session; err: %v", err),
```

Comment:
If we have the `err` object, let's just return it as it is without any formatting. We only return a custom error when it's a bifrost-related error. 
The formatting is required because we already created error codes for users to harness if needed.